crates/sui-config/src/builder.rs

```diff
@@ -276,6 +276,7 @@ impl<R: ::rand::RngCore + ::rand::CryptoRng> ConfigBuilder<R> {
                     consensus_config: Some(consensus_config),
                     enable_event_processing: false,
                     enable_gossip: true,
+                    gossip_degree: validator_num - 1,
```

Comment:
why - 1 ?

Reply:
because a validator does not follow itself :)

Reply:
A validator shouldn't need to follow N - 1 validators, which seems very wasteful. We may need some experimentation to figure out what would be the ideal number. Is there evidence that 4 is not enough?

Reply:
yeah we should do some experiments. Technically it should be 1/3 of the committee size to be effective.

Reply:
It certainly cannot be N - 1. We don't want O(n^2) gossip messages.

The correct degree setting is going to be a matter of balancing likelihood of any validator falling behind vs cost of gossip. Unless anyone here has a phd in graph theory I suspect we will need to do some experiments or simulations to determine the best setting.

At the very least, it would be good to see some evidence that 4 is not enough.